packages/get-starknet/src/rpcs/add-invoke.ts

```diff
@@ -0,0 +1,17 @@
+import type { RpcTypeToMessageMap } from 'get-starknet-core';
+
+import { StarknetWalletRpc } from '../utils/rpc';
+
+export type WalletAddInvokeTransactionMethod = 'wallet_addInvokeTransaction';
+type Params = RpcTypeToMessageMap[WalletAddInvokeTransactionMethod]['params'];
+type Result = RpcTypeToMessageMap[WalletAddInvokeTransactionMethod]['result'];
+
+export class WalletAddInvokeTransaction extends StarknetWalletRpc {
+  async handleRequest(params: Params): Promise<Result> {
+    const { calls } = params;
+
+    const result = await this.snap.addInvoke(this.wallet.selectedAddress, calls);
```

Comment:
In general this workflow should move to RPC, but get-starknet does not use that error for some API, so we keep this here (TEMP)
```suggestion
    try {
        return await this.snap.execute(this.wallet.selectedAddress, formatCalls(calls));
    } catch (error) {
        // if it is valdation Error, we mask it to get-starknet error
        if (Object.prototype.hasOwnProperty.call(error, "code") && error.code === MetaMaskErrorCode.ValidationError) {
            // Use the validation message from snap
            throw new WalletRpcError(error.message, WalletRpcErrorCode.InvalidPayload);
        }
        // if this is not a validation error, just throw the error, 
        // and the parent class will convert it to starknet error
        throw error;
    }
```

put this in constants.ts
```
enum MetaMaskErrorCode {
    ValidationError = "-32602"
}
```

WalletRpcErrorCode.InvalidPayload in error.ts
```
export enum WalletRpcErrorCode {
  InvalidPayload = 114,
}
```